processor/src/main/java/org/mapstruct/ap/internal/model/NormalTypeMappingMethod.java

```diff
@@ -89,6 +89,10 @@ public boolean equals(Object obj) {
         }
         NormalTypeMappingMethod other = (NormalTypeMappingMethod) obj;
 
+        if ( !super.equals( obj ) ) {
```

Comment:
This was a bug that was not calling the super equals methods and thus the `parameters` and the `returnType` was not taken into consideration for the equals methods.